presto-geospatial/src/main/java/com/facebook/presto/plugin/geospatial/GeoFunctions.java

```diff
@@ -384,33 +383,15 @@ public static Slice stCentroid(@SqlType(GEOMETRY_TYPE_NAME) Slice input)
             return serialize(createFromEsriGeometry(new Point(), geometry.getEsriSpatialReference()));
         }
 
-        Point centroid;
         try {
-            switch (geometryType) {
-                case MULTI_POINT:
-                    centroid = computePointsCentroid((MultiVertexGeometry) geometry.getEsriGeometry());
-                    break;
-                case LINE_STRING:
-                case MULTI_LINE_STRING:
-                    centroid = computeLineCentroid((Polyline) geometry.getEsriGeometry());
-                    break;
-                case POLYGON:
-                    centroid = computePolygonCentroid((Polygon) geometry.getEsriGeometry());
-                    break;
-                case MULTI_POLYGON:
-                    centroid = computeMultiPolygonCentroid((OGCMultiPolygon) geometry);
-                    break;
-                default:
-                    throw new PrestoException(INVALID_FUNCTION_ARGUMENT, "Unexpected geometry type: " + geometryType);
-            }
+            return serialize(geometry.centroid());
         }
         catch (RuntimeException e) {
             if (e instanceof PrestoException && ((PrestoException) e).getErrorCode() == INVALID_FUNCTION_ARGUMENT.toErrorCode()) {
                 throw e;
             }
             throw new PrestoException(INVALID_FUNCTION_ARGUMENT, format("Cannot compute centroid: %s Use ST_IsValid to confirm that input geometry is valid or compute centroid for a bounding box using ST_Envelope.", e.getMessage()), e);
```

Comment:
@jagill Is this logic still needed? I see that you removed the test that triggered it.

Reply:
I suppose not!